internal/controller/vrg_volsync.go

```diff
@@ -347,16 +350,22 @@ func (v *VRGInstance) aggregateVolSyncDataReadyCondition() *metav1.Condition {
 		Message:            "All VolSync PVCs are ready",
 	}
 
-	if v.instance.Spec.ReplicationState == ramendrv1alpha1.Secondary {
+	if len(v.volSyncPVCs) == 0 {
 		dataReadyCondition.Reason = VRGConditionReasonUnused
-		dataReadyCondition.Message = "Volsync based PVC protection does not report DataReady condition as Secondary"
+		dataReadyCondition.Message = "No PVCs are protected using Volsync scheme"
 
 		return dataReadyCondition
 	}
 
-	if len(v.volSyncPVCs) == 0 {
-		dataReadyCondition.Reason = VRGConditionReasonUnused
-		dataReadyCondition.Message = "No PVCs are protected using Volsync scheme"
+	if v.instance.Spec.ReplicationState == ramendrv1alpha1.Secondary {
+		if v.volSyncHandler.GetWorkloadStatus() == "active" {
+			dataReadyCondition.Status = metav1.ConditionFalse
+			dataReadyCondition.Reason = VRGConditionReasonProgressing
+			dataReadyCondition.Message = "Some of the VolSync PVCs are active"
+		}
+
+		dataReadyCondition.Reason = VRGConditionReasonReady
```

Comment:
Reason and message are overwritten here, the ones set in lines 363 and 364 are lost, we need to return right after 364 to preserve the reason and message when PVCs are active.

Reply:
This one needs to be addressed I think...